tests/unit/deepedit/test_transforms.py

```diff
@@ -12,30 +12,124 @@
+)
 
 IMAGE = np.array([[[[1, 0, 2, 0, 1], [0, 1, 2, 1, 0], [2, 2, 3, 2, 2], [0, 1, 2, 1, 0], [1, 0, 2, 0, 1]]]])
 LABEL = np.array([[[[0, 0, 0, 0, 0], [0, 1, 0, 1, 0], [0, 0, 1, 0, 0], [0, 1, 0, 1, 0], [0, 0, 0, 0, 0]]]])
-BATCH_IMAGE = np.array([[[[[1, 0, 2, 0, 1], [0, 1, 2, 1, 0], [2, 2, 3, 2, 2], [0, 1, 2, 1, 0], [1, 0, 2, 0, 1]]]]])
-BATCH_LABEL = np.array([[[[[0, 0, 0, 0, 0], [0, 1, 0, 1, 0], [0, 0, 1, 0, 0], [0, 1, 0, 1, 0], [0, 0, 0, 0, 0]]]]])
+MULTIMODALITY_IMAGE = np.random.rand(5, 5, 5)
+MULTI_LABEL = np.random.randint(0, 6, (5, 5))
 
 DATA_1 = {
     "image": IMAGE,
     "label": LABEL,
-    "image_meta_dict": {"dim": IMAGE.shape},
+    "image_meta_dict": {
+        "dim": np.array(IMAGE.shape),
+        "pixdim": [1, 1, 1, 5, 1, 1, 1, 1],
+        "filename_or_obj": "IMAGE_NAME",
+    },
     "label_meta_dict": {},
     "foreground": [0, 0, 0],
     "background": [0, 0, 0],
 }
 
+DISCARD_ADD_GUIDANCE_TEST_CASE = [
+    {"image": IMAGE, "label": LABEL},
+    DATA_1,
+    (3, 1, 5, 5),
+]
+
+DATA_2 = {
+    "image": IMAGE,
+    "label": LABEL,
+    "guidance": np.array([[[1, 0, 2, 2]], [[-1, -1, -1, -1]]]),
+    "discrepancy": np.array(
+        [
+            [[[[0, 0, 0, 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 0], [0, 0, 0, 0, 0], [0, 0, 0, 0, 0]]]],
+            [[[[0, 0, 0, 0, 0], [0, 0, 0, 0, 0], [0, 1, 0, 0, 0], [0, 0, 0, 0, 0], [0, 0, 0, 0, 0]]]],
+        ]
+    ),
+    "probability": 1.0,
+}
+
+CLICK_RATIO_ADD_RANDOM_GUIDANCE_TEST_CASE_1 = [
+    {"guidance": "guidance", "discrepancy": "discrepancy", "probability": "probability"},
+    DATA_2,
+    "[[[1, 0, 2, 2], [-1, -1, -1, -1]], [[-1, -1, -1, -1], [1, 0, 2, 1]]]",
+]
+
+DATA_3 = {
+    "image": np.arange(1000).reshape((1, 5, 10, 20)),
+    "image_meta_dict": {"foreground_cropped_shape": (1, 10, 20, 40), "dim": [3, 512, 512, 128]},
+    "guidance": [[[6, 10, 14], [8, 10, 14]], [[8, 10, 16]]],
+    "foreground": [[10, 14, 6], [10, 14, 8]],
+    "background": [[10, 16, 8]],
+}
+
+RESIZE_GUIDANCE_TEST_CASE_1 = [
+    {"ref_image": "image", "guidance": "guidance"},
+    DATA_3,
+    [[[0, 0, 0], [0, 0, 1]], [[0, 0, 1]]],
+]
+
+DATA_4 = {
+    "image": MULTIMODALITY_IMAGE,
+    "label": MULTI_LABEL,
+    "image_meta_dict": {
+        "dim": [1, 1, 1, 5, 2, 1, 1, 1],
+        "pixdim": [1, 1, 1, 5, 2, 1, 1, 1],
+        "filename_or_obj": "IMAGE_NAME",
+    },
+    "label_meta_dict": {
+        "dim": [1, 1, 1, 5, 2, 1, 1, 1],
+        "pixdim": [1, 1, 1, 5, 2, 1, 1, 1],
+        "filename_or_obj": "LABEL_NAME",
+    },
+}
+
+SINGLE_LABEL_SINGLE_MODALITY_TEST_CASE_1 = [
+    {"keys": ("image", "label")},
+    DATA_4,
+    (5, 5),
+]
+
+
+class TestDiscardAddGuidanced(unittest.TestCase):
+    @parameterized.expand([DISCARD_ADD_GUIDANCE_TEST_CASE])
+    def test_correct_results(self, arguments, input_data, expected_result):
+        add_fn = DiscardAddGuidanced(arguments)
+        result = add_fn(input_data)
+        self.assertEqual(result["image"].shape, expected_result)
```

Comment:
Is it possible to verify the contents of the output against expected values?

Reply:
Thanks for the comment! In this case verifying the content isn't necessary as the transform only add or discard tensors in the image. So it is more important to check shape than content.

Reply:
@diazandr3s thanks! I understand, the purpose of unittests is to check for the unexpected. What if the transform unexpectedly changes values in the remaining channel? Would that be covered? Also in this case, since the last two channels are discarded, you do need to check whether that operation was performed successfully or something else happened. To me checking contents would make a lot sense here.

Reply:
Many thanks @masadcv 
I just added that bit in this PR (https://github.com/Project-MONAI/MONAILabel/pull/391)